test/helpers/position_assertions.cc

```diff
@@ -525,14 +525,16 @@ void DefAssertion::check(const UnorderedMap<string, shared_ptr<core::File>> &sou
                         prettyPrintRangeComment(getLine(config, sourceFileContents, *missing), *missing->range, "")));
     }
 
-    for (auto &unexpected : diff.unexpected) {
-        ADD_FAIL_CHECK_AT(
-            locFilename.c_str(), line + 1,
-            fmt::format(
-                "Sorbet reported unexpected definition for location. Definition at:\n{}\nReported an "
-                "unexpected (additional?) definition at:\n{}",
-                prettyPrintRangeComment(locSourceLine, *makeRange(line, character, character + 1), ""),
-                prettyPrintRangeComment(getLine(config, sourceFileContents, *unexpected), *unexpected->range, "")));
+    if (diff.unexpected.size() > 1) {
+        for (auto &unexpected : diff.unexpected) {
+            ADD_FAIL_CHECK_AT(
+                locFilename.c_str(), line + 1,
+                fmt::format(
+                    "Sorbet reported unexpected definition for location. Definition at:\n{}\nReported an "
+                    "unexpected (additional?) definition at:\n{}",
+                    prettyPrintRangeComment(locSourceLine, *makeRange(line, character, character + 1), ""),
+                    prettyPrintRangeComment(getLine(config, sourceFileContents, *unexpected), *unexpected->range, "")));
+        }
```

Comment:
iiuc, this just silences the error, right?

Reply:
is this still needed when the result of the `T.let` is discarded?

Reply:
I'll try reverting this commit and checking. It would be really nice to not need to change these tests at all.

Reply:
Those tests still fail with this commit reverted, so I think that I'll need to wait for jvilk to address this.

Reply:
What's failing? Let's chat when you get back.